src/Mealz/MealBundle/Service/ParticipationService.php

```diff
@@ -65,6 +68,34 @@ public function join(Profile $profile, Meal $meal, ?Slot $slot = null, array $di
         return null;
     }
 
+    /**
+     * Update combined meal dishes for a participant.
+     *
+     * @param string[] $dishSlugs
+     *
+     * @throws ParticipationException
+     */
+    public function updateCombinedMeal(Participant $participant, array $dishSlugs): void
+    {
+        if (!$participant->getMeal()->isOpen()) {
+            throw new ParticipationException(
+                'expired participation; meal update not allowed',
+                ParticipationException::ERR_PARTICIPATION_EXPIRED
+            );
+        }
+        if ($this->isParticipationLocked($participant)) {
```

Comment:
Check could be moved to Meal like "isOpen".

Reply:
I also thought about the same. But logically it's the participation that is locked and not the meal. We only update participant.

Reply:
Hmm, you could say the same for is open. In my opinion it depends on the name of the method: isParticipationLocked is clearly for participation (e.g. isOpenForParticipation). But if you say isLocked and isOpen the check could be inside of Meal. So I would keep it at one place.

Reply:
I see the meal as open and expired. Participation can be locked or unlocked/open.
Furthermore, the information about participation should stay in participation service. Meal shouldn't be aware of participation. It doesn't belong to its domain in my opinion.

Reply:
What do you mean by 'Participation can be locked or unlocked/open.'? Participation is possible when a condition is met. The information that the meal has a lock time is IN the meal itself. I see it as state of the meal: Meal open, Meal Locked, Meal in the past (not opened, not locked). 

Well, I don't wanna fight. Do what you think is correct. In my opinion it's strange to say isOpen is here but isLocked is in another place. I just vote for the same place because as you see in my other comment isOpen and isLocked are connected.

Reply:
Please check also:
- ParticipationService: allowedToAccept similar check (and correct) check for meals
- MealCollection where participation is not involved: you check meal for LockDateTime and if it is open
- ParticipationServiceTrait: mealIsBookable similar check

So, you find (almost) the same thing in a lot of places which makes it very confusing.

Reply:
Well, I'm also putting my opinion and that is what code reviews are for. 
Logically, meal is just a meal. There is nothing to open or lock there. 
It's true that the information about participation is stored in meal, but it was a mistake that we did in past. We should at least try not to further expand it, if we can't fix it.

Reply:
I don't think that it was a (big) mistake. You can't save it in the participation. A meal is a dish at a certain time for an amount of money that a user can book. If a user participates/bookes a meal, meal tells you when it happens, what price it has, also the limit and when it will be locked. 

isBookable / isParticipationPossible should be (of course) a check in the participation service!
isOpen, isLocked could be in the participation service or in meal but not separated... isOpen takes the current time... You can also discuss about why meal knows something about the presence. :)

Reply:
In my opinion lock date-time is redundant. It can be easily be derived from the meal date. This way it be clean, concise of free of all duplicate logic that we currently have in multiple places.

Reply:
+1 (hopefully in the future!)